includes/gateway/class-omise-payment-fbbot.php

```diff
@@ -0,0 +1,266 @@
+			$messenger_id = $wp_query->query_vars['messenger_id'];
+
+			$url = plugin_dir_path( dirname( __DIR__ ) ) . 'templates/fbbot/payment-form.php';
+			ob_start();
+			include( $url );
+			return ob_get_clean();
+		}
+
+		public function payment_error_page_render () {
+			global $wp_query;
+
+			if ( ! isset( $wp_query->query_vars['error_message'] ) ) {
+			  return '<strong>Woocommerce system has error. Please try again.</strong>';
+			}
+
+			$error_message = $wp_query->query_vars['error_message'];
+
+			return '<strong>' . $error_message . '</strong>';
+		}
+
+		public function process_payment_by_bot( $params ) {
+		    /* rearrange step
+		      1. receive post data from payment page
+		      2. Create wc order
+		      3. Create charge 
+		      4. update order status
+		    */
+
+		    $omise_token = $params['omise_token'];
+		    $product_id = $params['product_id'];
+		    $messenger_id = $params['messenger_id'];
+
+		    $order = Omise_FBBot_WooCommerce::create_order( $params );
+
+		    $metadata = array(
+				'source' => 'woo_omise_bot',
+				'product_id' => $product_id,
+				'messenger_id' => $messenger_id,
+				'order_id' => $order->get_order_number()
+		    );
+
+		    $data = array(
+				'amount'      => $this->format_amount_subunit( $order->get_total(), $order->get_currency() ),
+				'currency'    => $order->get_currency(),
+				'description' => 'OrderID is '.$order->get_id().' : This order created from Omise FBBot and CustomerID is '.$messenger_id,
+				'metadata' => $metadata,
+				'card' => $omise_token
+		    );
+
+		    if ( $this->omise_3ds ) {
+		    	$return_uri =  site_url() . '/complete-payment';
+
+		      	$data['return_uri'] = $return_uri;
+		    }
+
+		    // Create Charge
+		    try {
+		      	$charge = OmiseCharge::create( $data, '', $this->secret_key() );
+		      	// We move checking charge status to request handler in handle triggered from omise method
+
+		      	// Just sent message to user for let them know we received these order
+		      	$prepare_confirm_message = Omise_FBBot_Conversation_Generator::prepare_confirm_order_message( $order->get_order_number() );
+		      	$response = Omise_FBBot_HTTPService::send_message_to( $messenger_id, $prepare_confirm_message );
+
+		      	// If merchant enable 3ds mode
+		      	if ( isset ( $charge['authorize_uri'] ) ) {
+		        	if ( wp_redirect( $charge['authorize_uri'] ) ) {
+		          		error_log( 'redirect to -> '. $charge['authorize_uri'] );
+		          		exit;
+		        	}
+		        
+		        	return;
+		      	}
+
+		      	// If merchant disable 3ds mode : normal mode
+		      	$redirect_uri =  site_url() . '/complete-payment';
+		      	if ( wp_redirect( $redirect_uri ) ) {
+		          	exit;
+		      	}
+
+		    } catch (Exception $e) {
+		      	error_log("catch error : " . $e->getMessage());
+
+		      	$error_message = str_replace(" ", "%20", $e->getMessage());
+
+		      	// [WIP] - Redirect to error page
```

Comment:
@igroomgrim Is this still WIP? (just curious, saw you add a line to redirect already)